reconcile/test/test_terraform_tgw_attachments.py

```diff
@@ -0,0 +1,308 @@
+        "uid": account_tgw_connection["account"]["uid"],
+        "terraformUsername": account_tgw_connection["account"]["terraformUsername"],
+        "assume_role": assume_role,
+        "assume_region": cluster_with_tgw_connection["spec"]["region"],
+        "assume_cidr": cluster_with_tgw_connection["network"]["vpc"],
+    }
+
+    mocks["ts"].populate_additional_providers.assert_called_once_with(
+        [expected_tgw_account, expected_tgw_account]
+    )
+    mocks["ts"].populate_tgw_attachments.assert_called_once_with(
+        [
+            {
+                "connection_provider": "account-tgw",
+                "connection_name": f"{account_tgw_connection['name']}_{expected_tgw_account['name']}-{tgw['tgw_id']}",
+                "requester": {
+                    "tgw_id": tgw["tgw_id"],
+                    "tgw_arn": tgw["tgw_arn"],
+                    "region": tgw["region"],
+                    "routes": tgw["routes"],
+                    "rules": tgw["rules"],
+                    "hostedzones": tgw["hostedzones"],
+                    "cidr_block": account_tgw_connection["cidrBlock"],
+                    "account": expected_tgw_account,
+                },
+                "accepter": {
+                    "vpc_id": vpc_details["vpc_id"],
+                    "region": cluster_with_tgw_connection["spec"]["region"],
+                    "cidr_block": cluster_with_tgw_connection["network"]["vpc"],
+                    "route_table_ids": vpc_details["route_table_ids"],
+                    "subnets_id_az": vpc_details["subnets_id_az"],
+                    "account": expected_tgw_account,
+                },
+                "deleted": account_tgw_connection["deleted"],
+            }
+        ]
+    )
+
+
+def test_run_when_cluster_with_mixed_connections(
+    mocker,
+    cluster_with_mixed_connections,
+    account_tgw_connection,
+    tgw,
+    vpc_details,
+    assume_role,
+):
+    mocks = _setup_mocks(
+        mocker,
+        clusters=[cluster_with_mixed_connections],
+        vpc=(
+            vpc_details["vpc_id"],
+            vpc_details["route_table_ids"],
+            vpc_details["subnets_id_az"],
+        ),
+        tgws=[tgw],
+        assume_role=assume_role,
+    )
+
+    from reconcile.terraform_tgw_attachments import run
+
+    run(True)
+
+    expected_tgw_account = {
+        "name": account_tgw_connection["account"]["name"],
+        "uid": account_tgw_connection["account"]["uid"],
+        "terraformUsername": account_tgw_connection["account"]["terraformUsername"],
+        "assume_role": assume_role,
+        "assume_region": cluster_with_mixed_connections["spec"]["region"],
+        "assume_cidr": cluster_with_mixed_connections["network"]["vpc"],
+    }
+
+    mocks["ts"].populate_additional_providers.assert_called_once_with(
+        [expected_tgw_account, expected_tgw_account]
+    )
+    mocks["ts"].populate_tgw_attachments.assert_called_once_with(
```

Comment:
can we validate the the returns as well ?

Reply:
`populate_tgw_attachments` has no `return`